pkg/loki/modules.go

```diff
@@ -147,41 +147,40 @@ func (t *Loki) initRing() (_ services.Service, err error) {
 	return t.ring, nil
 }
 
-func (t *Loki) initGroupcache() (_ services.Service, err error) {
-	if !t.Cfg.Common.GroupCacheConfig.Enabled {
+func (t *Loki) initEmbeddedCache() (_ services.Service, err error) {
+	if !t.Cfg.QueryRange.CacheConfig.Embeddedcache.IsEnabledWithDistributed() {
 		return nil, nil
 	}
 
-	t.Cfg.Common.GroupCacheConfig.Ring.ListenPort = t.Cfg.Common.GroupCacheConfig.ListenPort
-	rm, err := cache.NewGroupcacheRingManager(t.Cfg.Common.GroupCacheConfig, util_log.Logger, prometheus.DefaultRegisterer)
+	groupConfig := cache.GroupCacheConfig{
+		Enabled:    true,
+		Ring:       t.Cfg.QueryRange.CacheConfig.Embeddedcache.Ring,
```

Comment:
This sets up the global groupcache and the config shouldn't come from the QueryRange Cache. If we go this route, enabling a chunks cache means that the chunks cache depends on the query cache config